backend/app/instance_provider.go

```diff
@@ -0,0 +1,60 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+)
+
+// InstanceFactoryFunc factory method for creating data source instances.
+type InstanceFactoryFunc func(settings backend.AppInstanceSettings) (instancemgmt.Instance, error)
+
+// NewInstanceManager creates a new data source instance manager,
+//
+// This is a helper method for calling NewInstanceProvider and creating a new instancemgmt.InstanceProvider,
+// and providing that to instancemgmt.New.
+func NewInstanceManager(fn InstanceFactoryFunc) instancemgmt.InstanceManager {
+	ip := NewInstanceProvider(fn)
+	return instancemgmt.New(ip)
+}
+
+// NewInstanceProvider create a new data source instance provuder,
+//
+// The instance provider is responsible for providing cache keys for data source instances,
+// creating new instances when needed and invalidating cached instances when they have been
+// updated in Grafana.
+// Cache key is based on the numerical data source identifier.
+// If fn is nil, NewInstanceProvider panics.
+func NewInstanceProvider(fn InstanceFactoryFunc) instancemgmt.InstanceProvider {
```

Comment:
Please fix comments above: s/data source/app and adapt to used cache key

Reply:
Done!